UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs

```diff
@@ -1319,13 +1319,20 @@ private void AddToSurface(ReagentMix Chemicals, ReagentMix mix)
 		[Server]
 		public virtual void OnGib()
 		{
-			_ = SoundManager.PlayAtPosition(CommonSounds.Instance.Slip, gameObject.transform.position,
+			SoundManager.PlayAtPosition(CommonSounds.Instance.Slip, gameObject.transform.position,
 				gameObject); //TODO: replace with gibbing noise
-			CirculatorySystem.Bleed(GetTotalBlood());
+			CirculatorySystem.OrNull()?.Bleed(GetTotalBlood());
 			Death();
+			StartCoroutine(RemoveAllBodyParts());
+		}
+
+		private IEnumerator RemoveAllBodyParts()
```

Comment:
why does this need to be a Coroutine?

Reply:
Ignore this, was testing something. Removed.